drivers/spi/spi_mcux_lpspi.c

```diff
@@ -276,7 +283,52 @@ static void spi_mcux_dma_callback(const struct device *dev, void *arg,
 			data->status_flags |= SPI_MCUX_LPSPI_DMA_ERROR_FLAG;
 		}
 	}
+#if CONFIG_SPI_ASYNC
+	if (data->ctx.asynchronous) {
+		const struct spi_mcux_config *config = spi_dev->config;
+
+		if ((data->status_flags & SPI_MCUX_LPSPI_DMA_DONE_FLAG)  ==
+			SPI_MCUX_LPSPI_DMA_DONE_FLAG) {
+			while ((LPSPI_GetStatusFlags(config->base) & kLPSPI_ModuleBusyFlag)) {
+			/* wait until module is idle */
+			}
+
+			/* Disable DMA */
+			LPSPI_DisableDMA(config->base, kLPSPI_TxDmaEnable | kLPSPI_RxDmaEnable);
+			/* Load dma blocks of equal length */
+			size_t dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
+
+			if (dma_size == 0) {
+				dma_size = MAX(data->ctx.tx_len, data->ctx.rx_len);
+			}
+
+			spi_context_update_tx(&data->ctx, 1, dma_size);
+			spi_context_update_rx(&data->ctx, 1, dma_size);
+
+			if (data->ctx.tx_len == 0 && data->ctx.rx_len == 0) {
+				spi_context_complete(&data->ctx, spi_dev, 0);
+			} else {
+				/* transceive needs to take in spi_buf_set */
+				const struct spi_buf_set tx = {
+					.buffers = data->ctx.current_tx,
+					.count = 1
+				};
+				const struct spi_buf_set rx = {
+					.buffers = data->ctx.current_rx,
+					.count = 1
+				};
+
+				transceive_dma(spi_dev, data->ctx.config,
```

Comment:
could we avoid calling `transceive_dma` here? There are a few reasons I'd prefer this:
- we can avoid creating the `spi_buf` set, which `transceive_dma` won't use in asynchronous mode anyways
- we can avoid the requirement to setup the SPI context buffers in `spi_mcux_transceive_async`, which I currently find slightly confusing (since it is normally done in `transceive_dma`)

Could we instead call `spi_mcux_dma_rxtx_load` from this function?

Reply:
Appreciate it, I completely overlooked this approach.